pathutil/pathutil.go

```diff
@@ -10,64 +10,54 @@ import (
 	"strings"
 )
 
-// TempDirProvider ...
-type TempDirProvider interface {
-	CreateTempDir(prefix string) (string, error)
+//
+// Path provider functions
+
+// PathProvider ...
+type PathProvider interface {
```

Comment:
You've mentioned this has been renamed to allow adding more functions in the future that are not necessarily related to temp dirs. What are those use cases?

Reply:
The functions between `// Path provider functions` and `// Path checker functions` comments:
- `NormalizedOSTempDirPath` (already added as `TempDir`)
- `CurrentWorkingDirectoryAbsolutePath`
- `UserHomeDir`
- `EnsureDirExist`

I feel like these functions are used to get some sort of file path.

Reply:
Let's move forward with this naming scheme then for now. 👍